enhancements/cluster-constraints-properties.md

```diff
@@ -0,0 +1,172 @@
+---
+title: cluster constraint and property
+authors:
+  - "@dinhxuanvu"
+reviewers:
+  - "@kevinrizza"
+  - "@joelandford"
+approvers:
+  - "@kevinrizza"
+creation-date: 2021-10-08
+last-updated: 2020-10-11
+status: provisional
+---
+
+# cluster-constraint-and-property
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The dependency resolution in OLM needs to work with cluster constraints in order to effectively install operators that will work properly in the cluster at runtime. Plus, OLM needs to support cluster properties which reflects the runtime information that the cluster may provide and allows operators to depend on those properties if needed to ensure proper installation and operation.
+
+## Motivation
+
+### Existing dependency resolution
+
+OLM currently only resolves constraints that are provided by the operators and the resolution outcome at runtime is solely based on the constraints being satisfied by other operators who provide the required properties. It is not possible at the moment for OLM to consider cluster information or properties during resolution or apply cluster constraints to the available operators to prevent or allow certain operators to be installed successfully.
+
+### Cluster information
+
+Operators can depend on certain information such as usable API(s) that are available in the cluster. It is important to surface those information so operators can determine at runtime if they will be able to install and work properly in the cluster. The cluster information/properties can be depended upon as a constraint for a particular operator. This dependency relationship will ensure only applicable operators can be installed in the cluster and prevent possible failed installation and potentially nonworking condition.
+
+### Cluster restriction
+
+The cluster itself can impose certain restriction to which operators can be installed in the cluster via the properties associated with the operators. For example, the cluster imposes a minumum kubernetes version all for installable operators. A constraint can be constructed for a specific cluster restriction and then will be resolved during dependency resolution to ensure only operators with corresponding property with a satisfiable value.
```

Comment:
I think it is more a "wording" thing but it is not the cluster itself that "imposes a minimum kubernetes version". It is the operator author that requires it for the operator to work. 
I think it is important in the context of "cluster constraints" to have a clear separation:
- The cluster has properties
- Operators may have constraints on these properties